test/extensions/filters/http/cache/simple_http_cache/simple_http_cache_test.cc

```diff
@@ -256,6 +275,29 @@ TEST_F(SimpleHttpCacheTest, VaryResponses) {
   EXPECT_TRUE(expectLookupSuccessWithBody(first_value_vary.get(), Body1));
 }
 
+TEST_F(SimpleHttpCacheTest, PutGetWithTrailers) {
+  const std::string request_path1("Name");
+  LookupContextPtr name_lookup_context = lookup(request_path1);
+  EXPECT_EQ(CacheEntryStatus::Unusable, lookup_result_.cache_entry_status_);
+
+  Http::TestResponseHeaderMapImpl response_headers{{"date", formatter_.fromTime(current_time_)},
+                                                   {"cache-control", "public,max-age=3600"}};
+  const std::string body1("Value");
+  Http::TestResponseTrailerMapImpl response_trailers{{"why", "is"}, {"sky", "blue"}};
+  insert(move(name_lookup_context), response_headers, body1, response_trailers);
+  name_lookup_context = lookup(request_path1);
+  EXPECT_TRUE(expectLookupSuccessWithBody(name_lookup_context.get(), body1, response_trailers));
+
+  const std::string& request_path2("Another Name");
+  LookupContextPtr another_name_lookup_context = lookup(request_path2);
+  EXPECT_EQ(CacheEntryStatus::Unusable, lookup_result_.cache_entry_status_);
+
+  const std::string new_body1("NewValue");
+  insert(move(name_lookup_context), response_headers, new_body1, response_trailers);
+  EXPECT_TRUE(
+      expectLookupSuccessWithBody(lookup(request_path1).get(), new_body1, response_trailers));
```

Comment:
See above comment about leaving expectLookupSuccessWithBody as it was, and testing the trailers separately. In this case you'd need a temp for the lookup result so there'd be one extra line:
```
LookupContextPtr lookup_context = lookup(request_path);
expectLookupSuccessWithBody(lookup_context.get(), new_body);
EXPECT_EQ(response_trailers, getTrailers(*lookup_context));
```

Same with the call-site on line 289 though in that case there'd be 2 extra lines.

Reply:
Sorry, this comment is obsolete; I started to make this suggestion in another comment but changed my mind, and then forgot to delete this comment.